src/Connection.php

```diff
@@ -506,11 +506,7 @@ public function setAutoCommit($autoCommit)
      */
     public function fetchAssociative(string $query, array $params = [], array $types = [])
     {
-        try {
-            return $this->executeQuery($query, $params, $types)->fetchAssociative();
-        } catch (Driver\Exception $e) {
```

Comment:
The above methods cannot throw a driver exception. There's an exception converter involved everywhere which will wrap the driver exception in a wrapper.